### DISCUSSION TITLE
General Slowness of Server Responses

### DISCUSSION THREAD

**Discussion by mikolaszko** [livewire/livewire#6182] [1. Help] [answered]
Howdy! 
I recently decided that I would like to build a PAAS on TALL stack and it's been quite fun until I got into the livewire performance issues.
It's my first time ever developing with technology that sends back all the html needed to keep the UI reactive and I am fine with some slow-downs but when it takes 8kB response over 1000ms I can't think but wonder if there's something wrong I am doing?
I am using PlanetScale as a MySql provider and I've checked all the queries, pretty much non of them exceed the 4ms mark so I don't think it's a DB problem. My production server is running on a small Digital Ocean instance but I don't see any increase in the response time regardless if I test on my local machine or on the DigitalOcean (its tad bit slower).
Here I attach the screenshot from my network tab:
![image](https://github.com/livewire/livewire/assets/88675628/ee109605-2046-4b99-b791-b20977f54c0e)
![image](https://github.com/livewire/livewire/assets/88675628/4fd6f597-68e0-442b-a652-16b0c53f953d)

and here is my livewire component (i've tried multiple different things but this seems like the only way to refresh the UI)
```
<?php
namespace App\Http\Livewire;

use App\Models\ToDo;
use Illuminate\Support\Facades\Auth;
use Livewire\Component;


class TodoSection extends Component
{
    public string $content;
    public $todayToDos;

    public function mount() {
        $this->refreshTodos();
    }

    private function refreshTodos() {
        $this->todayToDos = ToDo::where("created_at", date("d/m/y"))->get();
    }

    public function render()
    {
        return view('livewire.dashboard.todo-section', ['todos' => $this->todayToDos]);
    }
    public function addTodo()
    {
        ToDo::create(['content' => $this->content, 'userId' => Auth::id(), 'isFinished' => false, 'created_at' => date("d/m/y")]);
        $this->refreshTodos();
    }
    public function deleteTodo($id)
    {
        ToDo::where('id', $id)->delete();
        $this->refreshTodos();
    }
}
```

and here's my blade component

```
<div class="h-full flex flex-col dark:bg-sky-900 p-6 w-full">
    <div class="mb-4">
        <section class="flex justify-between w-full">
            <span class="text-stone-50">Todo List</span>
            <span>{{sizeof($todos)}} {{sizeof($todos) === 1 ? 'task' : 'tasks'}}</span>
        </section>
        <div class="flex justify-between mt-4">
            <section class="flex border rounded border-teal-400 mr-4">
                <input wire:model.lazy="content" class="w-full py-2 px-3 flex-1 text-gray-700" placeholder="Add Todo">
                <select class="py-2 px-3 text-gray-700 border-none min-w-[120px]"  placeholder="Add Todo">
                    <option selected>Category</option>
                    <option value="career">Career</option>
                    <option value="family">Family</option>
                    <option value="personal">Personal</option>
                    <option value="sport">Sport</option>
                </select>
            </section>
            <button wire:click="addTodo" class="flex-no-shrink p-2 border-2 rounded text-teal-300 border-teal-400 hover:text-white hover:bg-opacity-80">Add</button>
        </div>
    </div>
    <div class="h-full flex flex-col gap-y-2">
        @foreach ($todos as $todo)
            <section class="flex items-center justify-between border rounded-full border-teal-400 px-3" wire:key="task-{{$todo->id}}">
                <span>{{ $todo->content }}</span>
                <section class="flex gap-1">
                    <input type="checkbox" class="cursor-pointer h-4 w-4 text-teal-500 rounded border-none transition duration-150 ease-in-out">
                    <x-bi-archive wire:click="deleteTodo({{$todo->id}})" class="stroke-red-700 cursor-pointer"/>
                </section>
            </section>
        @endforeach
    </div>
</div>
```

The reason I wanted to use TALL stack because of all the JS fatigue I experience at work and I really want to learn and be proficient but I can't seem to make it work. 

Is the slow server response by design because Livewire was build to not handle a ton of user interactions? I seriously doubt that and if there's somebody that could help me, Id really appreciate it :hand:

**Comment by ildyria** [comment]
Have you tried to use a computed property instead of `public $todayToDos;` ?
This will allow you to skip all the dehydration/hydration phases when loading the component.

**Reply by mikolaszko** [reply]
Thanks for advice, I've taken a look at docs and I need more time to understand the idea behind computed properties :) 

I'll try to have a look at it this weekend but even if the issue is fixed - isn't it strange that it takes so long?

**Reply by mikolaszko** [reply]
It helped a bit but I still have to wait 600ms for the server to give me back the newly created to-do and render it out. It still not fast

**Reply by derekcavaliero** [reply]
@mikolaszko I've noticed my local server environment (I use MAMP Pro) is slow, but once I deploy to a production/staging server (Digital Ocean VPS) those speed issues seem to go away entirely. Unsure if this is just a byproduct of something w/ how localhost MySQL is setup compared to an actual production environment.

**Comment by joshhanley** [comment]
@mikolaszko definitely not by design! Not sure exactly what is going on, but responses definitely should be quicker than that. You may have to do some debugging and logging timestamps to see if you can narrow down exactly where the slowdown is. Laravel Debugbar or Clockwork might be able to help with that too.

**Reply by mikolaszko** [reply]
I was able to narrow it down to this query, for now I have no clue what triggers it but Ill look around :D 
![image](https://github.com/livewire/livewire/assets/88675628/b7e7c500-fd93-49eb-8acc-c5ee3b9b9163)
thanks for the Debugbar suggestion

**Reply by mikolaszko** [reply]
well actually it's beyond my understanding, removing the middleware auth prolonged other queries resulting in 
![image](https://github.com/livewire/livewire/assets/88675628/a1c5d032-4a38-495f-9eb2-16e29ce69717)
This is happening regardless of what I do meaning insert and delete are suddenly slow
@joshhanley do you have an idea of why this might be happening?

**Reply by ildyria** [reply]
I would also try clockwork: https://underground.works/clockwork/
It gives quite good insights of code paths, especially with livewire.
![image](https://github.com/livewire/livewire/assets/627094/f84fcd5e-6088-45e3-8455-1959c784531a)

**Reply by mikolaszko** [reply]
In my case the queries are taking forever to get a response, maybe there's a big latency between the PlanetScale and my server? this shouldnt really be the case since PS performs them almost instantly but Ill try other stuff out
![image](https://github.com/livewire/livewire/assets/88675628/513ec7c5-b27f-4328-ba2a-a92b117c2db1)

**Reply by mikolaszko** [reply]
@ildyria in the example app that you provided are you using an external db provider?

**Reply by sbc640964** [reply]
Trouble with the first query
I am also experiencing the problem on many of my Laravel apps
The first query takes an excessive amount of time and not relative to the others,
It doesn't matter what it is, it will always take much longer - a lot of time.
Let me know if you find a way to deal with the problem.

**Reply by ildyria** [reply, marked as answer]
> @ildyria in the example app that you provided are you using an external db provider?

A local MySQL DB.

**Reply by mikolaszko** [reply]
Raw SQL query didn't help but tinker manipulates/fetches data instantly

**Reply by romansh** [reply]
I had a performance issue with (if I am not wrong) DELETE and UPDATE queries unless I moved the folder with the database files to an SSD.

**Reply by mikolaszko** [reply]
I am not self hosting

**Comment by mikolaszko** [comment]
![image](https://github.com/livewire/livewire/assets/88675628/65e8506b-d718-462f-aec5-47685baaa082)
this is the time of querying after I switched from a PlanetScale DB to a local connection. Turns out Livewire wasn't at fault at all :sweat_smile: 
Thanks for help @ildyria @joshhanley  and I hope @sbc640964 you can find a solution
